drivers/display/display_ili9806e_dsi.c

```diff
@@ -0,0 +1,440 @@
+{
+	const struct ili9806e_config *cfg = dev->config;
+	int ret;
+
+	ret = ili9806e_write_sequence(dev, init_cmds, ARRAY_SIZE(init_cmds));
+	if (ret < 0) {
+		return ret;
+	}
+	/* Add a delay, otherwise MADCTL not taken */
+	k_msleep(120);
+
+	/* Exit sleep mode */
+	ret = ili9806e_write_reg(dev, MIPI_DCS_EXIT_SLEEP_MODE, NULL, 0);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Wait for sleep out exit */
+	k_msleep(5);
+
+	/* Set color mode */
+	ret = ili9806e_write_reg_val(dev, MIPI_DCS_SET_PIXEL_FORMAT,
+				     cfg->pixel_format == PIXEL_FORMAT_RGB_565
+					     ? ILITEK_ILI9806E_COLMOD_RGB565
+					     : ILITEK_ILI9806E_COLMOD_RGB888);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Turn on display */
+	ret = ili9806e_write_reg(dev, MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+	return ret;
+}
+
+static int ili9806e_blanking_on(const struct device *dev)
+{
+	const struct ili9806e_config *cfg = dev->config;
+	int ret;
+
+	if (cfg->backlight.port != NULL) {
+		ret = gpio_pin_set_dt(&cfg->backlight, 0);
+		if (ret) {
+			LOG_ERR("Disable backlight failed! (%d)", ret);
+			return ret;
+		}
+	}
+
+	return ili9806e_write_reg(dev, MIPI_DCS_SET_DISPLAY_OFF, NULL, 0);
+}
+
+static int ili9806e_blanking_off(const struct device *dev)
+{
+	const struct ili9806e_config *cfg = dev->config;
+	int ret;
+
+	if (cfg->backlight.port != NULL) {
+		ret = gpio_pin_set_dt(&cfg->backlight, 1);
+		if (ret) {
+			LOG_ERR("Enable backlight failed! (%d)", ret);
+			return ret;
+		}
+	}
+
+	return ili9806e_write_reg(dev, MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+}
+
+static int ili9806e_write(const struct device *dev, uint16_t x, uint16_t y,
+			  const struct display_buffer_descriptor *desc, const void *buf)
+{
+	LOG_WRN("Write not supported, use LCD controller display driver");
+	return -ENOTSUP;
+}
+
+static void ili9806e_get_capabilities(const struct device *dev,
+				      struct display_capabilities *capabilities)
+{
+	const struct ili9806e_config *cfg = dev->config;
+
+	memset(capabilities, 0, sizeof(struct display_capabilities));
+	capabilities->x_resolution = cfg->width;
+	capabilities->y_resolution = cfg->height;
+	capabilities->supported_pixel_formats = cfg->pixel_format;
+	capabilities->current_pixel_format = cfg->pixel_format;
+}
+
+static int ili9806e_pixel_format(const struct device *dev,
+				 const enum display_pixel_format pixel_format)
+{
+	const struct ili9806e_config *config = dev->config;
+
+	if (pixel_format == config->pixel_format) {
+		return 0;
+	}
+	LOG_WRN("Pixel format change not implemented");
+
+	return -ENOTSUP;
+}
+
+static const struct display_driver_api ili9806e_api = {
+	.blanking_on = ili9806e_blanking_on,
+	.blanking_off = ili9806e_blanking_off,
+	.write = ili9806e_write,
+	.set_pixel_format = ili9806e_pixel_format,
+	.get_capabilities = ili9806e_get_capabilities,
+};
+
+static int ili9806e_init(const struct device *dev)
+{
+	const struct ili9806e_config *cfg = dev->config;
+	struct mipi_dsi_device mdev;
+	int ret;
+
+#if !CONFIG_INPUT_GT911
```

Comment:
This *cannot* be present here. Not every display module with a ili9806e_dsi controller will have a GT911 touch controller.

This looks like a workaround specific to the display module you are using. Could you go into more detail as to why this exists so we can figure out where to put this code? It looks like the ILI9806E and GT911 module might share a reset line? In that case is is possible to set init priorities so only the first device to initialize actually pulses the reset line?

Reply:
> It looks like the ILI9806E and GT911 module might share a reset line

It's correct. In this shield, first the gt911 has to be reset, then the ili9806e has been reset before and skips this reset handler code

Reply:
It is correct for the *shield*, not for the display controller. The issue here is that someone could build a module with a FT5336 touchscreen controller, and ILI9806E display controller- the GT911 and ILI9806E sharing a reset line is a decision specific to the *shield/module*.

If you need to have the GT911 driver pulse the reset line, then have the ILI9806E skip the reset pulse, simply do not define a reset GPIO for the ILI9806E devicetree node, and make sure the GT911 driver starts first, so it can pulse the reset line.